client/client.go

```diff
@@ -442,9 +443,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 		return nil, nil, err
 	}
 
+	isWait := false
+	if req != nil && req.URL != nil {
+		isWait, _ = strconv.ParseBool(req.URL.Query().Get("wait"))
+	}
+
 	var hctx context.Context
 	var hcancel context.CancelFunc
-	if c.headerTimeout > 0 {
+	if !isWait && c.headerTimeout > 0 {
```

Comment:
It copies but not flush in time. Copy only means put into memory I guess. But again, even if we fix our proxy, there are other proxies that do not flush header in time.